src/Raven.Client/Exceptions/UnsuccessfulFileAccessException.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+
+namespace Raven.Client.Exceptions
+{
+    public class UnsuccessfulFileAccessException : Exception
+    {
+        private static string GetMessage(string filePath) => 
+            $@"Failed to access a file at <{filePath}>. There are multiple possible reasons: {Environment.NewLine}
+* RavenDB process has insufficient permissions to access the file. Please verify OS file permissions. {Environment.NewLine}
```

Comment:
Let's add an enum that will specify what access is required (read or write) and put it here. It can be passed to the `UnsuccessfulFileAccessException` ctor. In both cases of the current usage we need to write, however in the future we might use this exception for other things.